includes/class-wcpdf-admin.php

```diff
@@ -567,32 +576,84 @@ public function get_bulk_actions() {
 	}
 
 	/**
-	 * Save invoice number
+	 * Save document number
 	 */
-	public function save_invoice_number_date($post_id, $post) {
-		$post_type = get_post_type( $post_id );
-		if( $post_type == 'shop_order' ) {
-			// bail if this is not an actual 'Save order' action
-			if ( ! isset($_POST['action']) || $_POST['action'] != 'editpost' ) {
-				return;
+	public function save_document_number_date_notes() {
+		if ( check_ajax_referer( "wpo_wcpdf_save_document", 'security', false ) === false ) {
+			wp_send_json_error( array(
+				'message' => 'nonce expired',
+			) );
+		}
+
+		if ( ! isset ($_POST['action'] ) || $_POST['action'] != 'wpo_wcpdf_save_document' ) {
+			wp_send_json_error( array(
+				'message' => 'bad action',
+			) );
+		}
+
+		if( empty( $_POST['order_id'] ) || empty( $_POST['document_type'] ) ) {
+			wp_send_json_error( array(
+				'message' => 'incomplete request',
+			) );
+		}
+
+		if ( ! current_user_can( 'manage_woocommerce' ) ) {
+			wp_send_json_error( array(
+				'message' => 'no permissions',
+			) );
+		}
+
+		$order_id      = absint( $_POST['order_id'] );
+		$order         = WCX::get_order( $order_id );
+		$document_type = sanitize_text_field( $_POST['document_type'] );
+
+		// parse form data
+		parse_str( $_POST['form_data'], $form_data );
+		if ( is_array( $form_data ) ) {
+			foreach ( $form_data as $key => &$value ) {
+				if ( is_array( $value ) && ! empty( $value[$order_id] ) ) {
+					$value = $value[$order_id];
+				}
 			}
+		}
 
-			$order = WCX::get_order( $post_id );
-			if ( $invoice = wcpdf_get_invoice( $order ) ) {
-				$_POST = stripslashes_deep( $_POST );
-				$document_data = $this->process_order_document_form_data( $_POST, $invoice->slug );
-				$invoice->set_data( $document_data, $order );
+		// remove legacy hooks (older versions of Pro)
+		if( function_exists( 'WPO_WCPDF_Pro' ) ) {
+			$pro_installed_version = get_option( 'wpo_wcpdf_pro_version' );
+			if ( version_compare( $pro_installed_version, '7.2.4', '<' ) && version_compare( $pro_installed_version, '7.2.2', '>' ) ) {
```

Comment:
what's the reason for this? We never released such versions, but maybe that's just a typo?

I actually think we should separate the processes of ajax saving order data (which should be handled like regenerate is currently) and regular order saving. It seems the latter doesn't work anymore, correct?